airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py

```diff
@@ -653,6 +654,14 @@ def request_params(
     def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
         yield from response.json().get(self.data_field, {}).get(self.name, [])
 
+    def backoff_time(self, response: requests.Response) -> Optional[float]:
+        rate_limit = response.headers.get("x-amzn-RateLimit-Limit", 0)
+        if rate_limit:
+            backoff_time = 1 / float(rate_limit)
+            return backoff_time
+        else:
+            return self.default_backoff_time
```

Comment:
Do you really think we should only make one request per minute by default 🤔 You mentioned in the description that a specific status is returned in case of rate throttling, do you think we could check the status value + the presence of the header?

Reply:
The documentation says that the default  rate limit (requests per second) is 0.0167 sor for me It make sense wait one minute to repeat the request when you get a 429 response status by default, that value can be higher for some merchants so if a rate limit is defined in the response we will use it.

About check the status value + the presence of rate limit in the header I think that is not a good idea. the base class `HttpStream` have already a logic implemented for that. we only need to overwrite the `backoff_time` you can see it here [http.py#L289](https://github.com/airbytehq/airbyte/blob/41dc82f0565fe0a7cd7ad56f5a5d7629c8f3b931/airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py#L289) 

But it could be a good idea to move this `backoff_time` function to an upper class so other streams (not only Orders)  can use it. what do you think? 🤔